src/main/java/org/apache/sysds/runtime/compress/CompressedMatrixBlockFactory.java

```diff
@@ -137,6 +138,25 @@ public static Pair<MatrixBlock, CompressionStatistics> compress(MatrixBlock mb,
 		return compress(mb, k, new CompressionSettingsBuilder(), root);
 	}
 
+	public static Pair<MatrixBlock, CompressionStatistics> compress(MatrixBlock mb, MatrixBlock sf, int k, WTreeRoot root) {
+		// Handle only row vectors, as column-wise quantization is not allowed.
+		// The restriction is handled upstream
+		LOG.debug("Number of columns of sf " + sf.getNumColumns());
+		double[] scaleFactors = new double[sf.getNumRows()];
```

Comment:
good solution to extract the double vector, 
However, in this case if the MatrixBlock is dense, please use sf.getDenseblockValues(), 
then you do not have to allocate anything.